backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/export/ImageRegistry.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.diagrams.export;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.eclipse.sirius.web.diagrams.Diagram;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Registers the images referenced in a {@link Diagram} for late retrieval as \<symbol\>.
+ *
+ * @author rpage
+ */
+public class ImageRegistry {
+    private final Logger logger = LoggerFactory.getLogger(ImageRegistry.class);
+
+    private HttpClient imageFetcher;
```

Comment:
I'm pretty sure that you could simply build your HttpClient here without having to pass it around all the code.